src/exodus/test/wallet_tests.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2019 The Zcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <boost/test/unit_test.hpp>
+
+#include "wallet/wallet.h"
+#include "wallet/test/wallet_test_fixture.h"
+
+#include "../wallet.h"
+
+class Wallet : public exodus::Wallet
+{
+public:
+    Wallet(const std::string& walletFile)
+        : exodus::Wallet(walletFile)
+    {
+    }
+
+    exodus::SigmaEntry GetSigmaEntry(const exodus::SigmaMintId& id)
+    {
+        return exodus::Wallet::GetSigmaEntry(id);
+    }
+
+    void ListSigmaEntries(std::list<exodus::SigmaEntry>& listSigma)
+    {
+        exodus::Wallet::ListSigmaEntries(std::back_inserter(listSigma));
+    }
+
+    void ListSigmaEntries(uint32_t propertyId, std::list<exodus::SigmaEntry>& listSigma)
+    {
+        exodus::Wallet::ListSigmaEntries(propertyId, std::back_inserter(listSigma));
+    }
+
+    bool HasSigmaEntry(const exodus::SigmaMintId& id)
+    {
+        return exodus::Wallet::HasSigmaEntry(id);
+    }
+};
+
+struct ExodusSigmaTestingSetup : WalletTestingSetup
+{
+    ExodusSigmaTestingSetup()
+    {
+        wallet = new Wallet(pwalletMain->strWalletFile);
+    }
+
+    ~ExodusSigmaTestingSetup()
+    {
+        delete wallet;
+        wallet = nullptr;
+    }
+    Wallet *wallet;
+
+    exodus::SigmaEntry createAndGetEntry(uint32_t propertyId, uint8_t denomination) {
+        auto pubKey = wallet->CreateSigmaMint(propertyId, denomination);
+        return wallet->GetSigmaEntry(
+            exodus::SigmaMintId(pubKey.GetCommitment(), propertyId, denomination)
+        );
+    }
+};
+
+BOOST_FIXTURE_TEST_SUITE(exodus_wallet_tests, ExodusSigmaTestingSetup)
+
+BOOST_AUTO_TEST_CASE(getid)
+{
+    exodus::SigmaEntry entry;
+    exodus::SigmaPrivateKey priv;
+    do {
+        priv.Generate();
+    } while (!priv.IsValid());
```

Comment:
If we make `SigmaPrivateKey` never generate invalid key we don't need this loop.